cmd/tailscaled/tailscaled.go

```diff
@@ -902,14 +886,30 @@ func applyIntegrationTestEnvKnob() {
 	}
 }
 
-func defaultEncryptState() bool {
+// defaultEncryptState reports whether we can enable state file encryption by
+// default, and returns an error if there's a concrete reason it can't be
+// enabled.
+func defaultEncryptState(statePath string) (bool, error) {
 	if runtime.GOOS != "windows" && runtime.GOOS != "linux" {
 		// TPM encryption is only configurable on Windows and Linux. Other
 		// platforms either use system APIs and are not configurable
 		// (Android/Apple), or don't support any form of encryption yet
 		// (plan9/FreeBSD/etc).
-		return false
+		return false, fmt.Errorf("--encrypt-state is not supported on %s", runtime.GOOS)
+	}
+	// Check if we have TPM support in this build.
+	if !store.HasKnownProviderPrefix(store.TPMPrefix + "/") {
```

Comment:
aside: weird names here. One thing takes a "prefix" and the other has a thing named "prefix" but isn't sufficient on its own so needs a "/" after it.